### ISSUE TITLE
1.5.0+ - in_forward fails on messages from td-agent with a compression error

### ISSUE THREAD

**Issue by 70k10** [fluent/fluent-bit#2396]
## Bug Report

**Describe the bug**
Fluent-bit 1.5.0+ does not accept forwarded messages from td-agent (tested with 3.8 & 4.0) regardless of compress value.
When td-agent is set to `compress text` (default), fluent-bit returns ` invalid 'compressed' option` and `gzip uncompress failure` when set to `compress gzip`.

Fluent-bit 1.4.6 works when td-agent is set to `compress text`, but fails when td-agent is set to `compress gzip` as expected.

I did some investigation within the v1.5.1 source and in the case of `compress text` the issue appeared to be within the fw_prot.c `is_gzip_compressed` function where the correct length message would fall through the compressed if statement and return -1.

**To Reproduce**
[systemd.log](https://github.com/fluent/fluent-bit/files/4982765/systemd.log)
[fluent-bit.conf.txt](https://github.com/fluent/fluent-bit/files/4982767/fluent-bit.conf.txt)
[td-agent.conf.txt](https://github.com/fluent/fluent-bit/files/4982766/td-agent.conf.txt)
[CMakeLists.txt](https://github.com/fluent/fluent-bit/files/4982770/CMakeLists.txt)

- Steps to reproduce the problem:
RHEL8/Centos8
Install development dependencies.
Compile fluent-bit with the attached CMakeLists.txt (I cannot install td-agent-bit internally):
```
cmake -D CPACK_GENERATOR:STRING=RPM ..
cpack -G RPM -C Release
```
Install fluent-bit (RPM in our case)
Copy systemd.log to /etc/td-agent/systemd.log
Copy conf files into their correct locations.
Run fluent-bit and td-agent.

Change `compress text` to `compress gzip` within td-agent.conf for additional issue.

**Expected behavior**
Fluent-bit outputs the forwarded messages using stdout.

**Your Environment**
* Version used: 1.5.1
* Configuration: See attached conf files
* Server type and version: VM
* Operating System and version: RHEL 8
* Filters and plugins: in_forward, out_stdout
I can provide an rpm list directly if necessary

**Additional context**
We are attempting to upgrade for the chunk changes to in_forward. It would also be nice to handle gzip compression due to the amount of logs we intend to push.

**Comment by edsiper** [comment]
Thanks for reporting the problem. 

Indeed the problem was the gzip uncompress routine used by the plugin, now it support all the variants of gzip. The fix is in commit e4fb0628edf127f82a7e9729615fe8cb414902df

note that the only compression mechanism supported by the protocol is ```gzip```.

**Comment by edsiper** [comment]
FYI: the fix will be included in v1.5.3 (date to be defined).

**Comment by 70k10** [comment]
Thanks for fixing this issue so quickly. I'll test the master branch and validate our workload. I'll have to dig into fluentd more to see what compress text is actually doing, maybe my other issue is a configuration one.
